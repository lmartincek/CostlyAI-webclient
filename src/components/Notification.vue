<template>
    <transition name="slide-fade">
        <div :class="['notification', type]">
            <span class="notification-message">{{ message }}</span>
        </div>
    </transition>
</template>

<script setup lang="ts">
defineProps({
    message: {
        type: String,
        required: true,
    },
    type: {
        type: String,
        default: 'success',
        validator: (value: string) => ['success', 'error', 'default'].includes(value),
    },
});
</script>

<style scoped lang="scss">
.notification {
    position: fixed;
    top: 20px;
    right: 50%;
    transform: translateX(50%);
    padding: 1rem 2rem;
    border-radius: 4px;
    color: $white-color;
    text-align: center;
    display: flex;
    align-items: center;
    gap: 12px;
    box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1);
    z-index: 1000;
    max-width: 350px;
    width: 100%;

    &.success {
        background-color: $success-color;
    }

    &.error {
        background-color: $error-color;
    }

    &.default {
        background-color: $headline-color;
    }
}

.notification-message {
    flex-grow: 1;
}

.notification-close {
    background: none;
    border: none;
    color: inherit;
    cursor: pointer;
    font-size: 16px;
    padding: 0;
    margin-left: 8px;
}

.slide-fade-enter-active,
.slide-fade-leave-active {
    transition: all 0.3s ease;
}

.slide-fade-enter-from,
.slide-fade-leave-to {
    opacity: 0;
    transform: translateX(100%);
}
</style>