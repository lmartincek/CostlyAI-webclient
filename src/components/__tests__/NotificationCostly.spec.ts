import { describe, it, expect, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import Notification from '@/components/common/NotificationCostly.vue'
import { nextTick } from 'vue'

describe('Notification Component', () => {
  const createWrapper = (props = {}) => {
    return mount(Notification, {
      props: {
        message: 'Test message',
        ...props,
      },
    })
  }

  describe('Rendering', () => {
    it('renders the notification when message is provided', () => {
      const wrapper = createWrapper()
      expect(wrapper.find('.notification').exists()).toBe(true)
      expect(wrapper.find('.notification-message').text()).toBe('Test message')
    })

    it('does not render when message is empty', () => {
      const wrapper = createWrapper({ message: '' })
      expect(wrapper.find('.notification').isVisible()).toBe(false)
    })
  })

  describe('Props Validation', () => {
    it('renders with default type (success)', () => {
      const wrapper = createWrapper()
      expect(wrapper.find('.notification').classes()).toContain('success')
    })

    it('applies correct class for error type', () => {
      const wrapper = createWrapper({ type: 'error' })
      expect(wrapper.find('.notification').classes()).toContain('error')
    })

    it('applies correct class for default type', () => {
      const wrapper = createWrapper({ type: 'default' })
      expect(wrapper.find('.notification').classes()).toContain('default')
    })

    it('throws warning for invalid type', () => {
      const spy = vi.spyOn(console, 'warn').mockImplementation(() => {})

      createWrapper({ type: 'invalid' })

      expect(spy).toHaveBeenCalled()
      spy.mockRestore()
    })
  })

  describe('Accessibility', () => {
    it('has proper aria-live attribute', () => {
      const wrapper = createWrapper()
      expect(wrapper.find('[aria-live="polite"]').exists()).toBe(true)
    })
  })

  describe('Transition Behavior', () => {
    it('applies transition classes', async () => {
      const wrapper = createWrapper()

      expect(wrapper.find('.notification').isVisible()).toBe(true)

      await wrapper.setProps({ message: '' })
      await nextTick()

      expect(wrapper.find('.slide-fade-leave-active').exists()).toBe(true)
    })
  })

  // describe('Snapshot Testing', () => {
  //   it('matches snapshot for success type', () => {
  //     const wrapper = createWrapper()
  //     expect(wrapper.html()).toMatchSnapshot()
  //   })
  //
  //   it('matches snapshot for error type', () => {
  //     const wrapper = createWrapper({ type: 'error' })
  //     expect(wrapper.html()).toMatchSnapshot()
  //   })
  // })
})
