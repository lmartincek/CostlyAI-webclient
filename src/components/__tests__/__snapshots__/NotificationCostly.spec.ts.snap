// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notification Component > Snapshot Testing > matches snapshot for error type 1`] = `"<div data-v-5a392ee7="" class="notification error" aria-live="polite"><span data-v-5a392ee7="" class="notification-message">Test message</span></div>"`;

exports[`Notification Component > Snapshot Testing > matches snapshot for success type 1`] = `"<div data-v-5a392ee7="" class="notification success" aria-live="polite"><span data-v-5a392ee7="" class="notification-message">Test message</span></div>"`;
